lxd/storage/drivers/driver_lvm_utils_test.go

```diff
@@ -0,0 +1,68 @@
+package drivers
+
+import (
+	"fmt"
+)
+
+func Example_lvm_parseLogicalVolumeName() {
+	d := &lvm{}
+	d.name = "pool"
+
+	type testVol struct {
+		lvName string
+		parent Volume
+	}
+
+	parentCT := Volume{
+		contentType: ContentTypeFS,
+		volType:     VolumeTypeContainer,
+		name:        "proj_testct-with-hyphens",
+	}
+
+	parentVM := Volume{
+		contentType: ContentTypeBlock,
+		volType:     VolumeTypeVM,
+		name:        "proj_testvm-with-hyphens",
+	}
+
+	custVol := Volume{
+		contentType: ContentTypeFS,
+		volType:     VolumeTypeCustom,
+		name:        "proj_testvol-with-hyphens.block", // .block ending doesn't indicate a block vol.
```

Comment:
@stgraber added a unit test to specifically test for the naming conflict scenario that will happen in the future when we add block support for custom volumes.

Reply:
@stgraber @monstermunchkin now we have custom block volumes, this might rear its head soon.